packages/ERTP/src/issuerKit.js

```diff
@@ -122,12 +122,9 @@ export const upgradeIssuerKit = (
   ) {
     Fail`Cannot (yet?) upgrade from 'noRecoverySets' to 'hasRecoverySets'`;
   }
-  if (
-    oldRecoverySetsState === 'hasRecoverySets' &&
-    recoverySetsOption === 'noRecoverySets'
-  ) {
-    Fail`Cannot (yet?) upgrade from 'hasRecoverySets' to 'noRecoverySets'`;
-  }
+  // if the existing state was 'hasRecoverySets' and recoverySetsOption is
+  // 'noRecoverySets', then we'll leave the old recoverySet in place, but not
+  // add to it.
```

Comment:
I had some difficulty making sense of this. It's not really about the option matrix, because `oldRecoverySetsState` is ignored if `recoverySetsOption == `noRecoverySets`.

```suggestion
  // Extant sets are never deleted. If the new option is 'noRecoverySets', they won't
  // be used but extant ones will remain.
```

Reply:
"Never" is too strong. The expectation is that we will find a way to delete the old objects in a controlled fashion, but we have to preserve the membership in the set in order to do so.

Reply:
Just for the record, https://github.com/Agoric/agoric-sdk/pull/8498 did such incremental deletion. Its deletion logic could be revived. If it is, the `'noRecoverySets'` option should be adequate to engage it, rather than requiring yet another options setting.

Reply:
When I last discussed this with @warner, he pointed out that those deletions and their consequences that cascade through the kernel to other vats are gated by GC passes, and  bring-out-your-dead phases. The effects of that buffering is that a deletions every so often in Zoe or some other vat might be saved up by the kernel and passed *en masse* to some other vat which might be overwhelmed. Until there's better support for regulating the flow of such updates between the kernel and downstream vats, he prefers that we not do mass deletions even at a slow rate.